src/software/SfM/main_SfMInit_ImageListing.cpp

```diff
@@ -407,38 +436,99 @@ int main(int argc, char **argv)
       focalPix = -1.0;
     }
 
-
     // Build intrinsic parameter related to the view
     std::shared_ptr<IntrinsicBase> intrinsic (NULL);
     if (focalPix > 0 && ppx > 0 && ppy > 0 && width > 0 && height > 0)
     {
+      EINTRINSIC camera_model = e_User_camera_model;
+      // If no user input choose a default camera model
+      if(camera_model == PINHOLE_CAMERA_START)
+      {
+        // Use standard lens with radial distortion by default
+        camera_model = PINHOLE_CAMERA_RADIAL3;
+        if(resizedImage)
+        {
+          // If the image has been resized, we assume that it has been undistorted
+          // and we use a camera without lens distortion.
+          camera_model = PINHOLE_CAMERA;
+        }
+        else if(focalLength_mm > 0.0 && focalLength_mm < 15)
+        {
+          // If the focal lens is short, the fisheye model should fit better.
+          camera_model = PINHOLE_CAMERA_FISHEYE;
+        }
+      }
+
       // Create the desired camera type
-      switch(e_User_camera_model)
+      switch(camera_model)
       {
         case PINHOLE_CAMERA:
+        {
           intrinsic = std::make_shared<Pinhole_Intrinsic>
             (width, height, focalPix, ppx, ppy);
-        break;
+          break;
+        }
         case PINHOLE_CAMERA_RADIAL1:
+        {
           intrinsic = std::make_shared<Pinhole_Intrinsic_Radial_K1>
             (width, height, focalPix, ppx, ppy, 0.0); // setup no distortion as initial guess
-        break;
+          break;
+        }
         case PINHOLE_CAMERA_RADIAL3:
+        {
           intrinsic = std::make_shared<Pinhole_Intrinsic_Radial_K3>
             (width, height, focalPix, ppx, ppy, 0.0, 0.0, 0.0);  // setup no distortion as initial guess
-        break;
+          break;
+        }
         case PINHOLE_CAMERA_BROWN:
-          intrinsic =std::make_shared<Pinhole_Intrinsic_Brown_T2>
+        {
+          intrinsic = std::make_shared<Pinhole_Intrinsic_Brown_T2>
             (width, height, focalPix, ppx, ppy, 0.0, 0.0, 0.0, 0.0, 0.0); // setup no distortion as initial guess
-        break;
+          break;
+        }
         case PINHOLE_CAMERA_FISHEYE:
-          intrinsic =std::make_shared<Pinhole_Intrinsic_Fisheye>
-            (width, height, focalPix, ppx, ppy, 0.0, 0.0, 0.0, 0.0); // setup no distortion as initial guess
-        break;
+        {
+          if(sCamName == "GoPro")
+          {
+            intrinsic = std::make_shared<Pinhole_Intrinsic_Fisheye>
+              (width, height, focalPix, ppx, ppy, 0.0524, 0.0094, -0.0037, -0.0004);
```

Comment:
not sure about using defaults for the distortion, need to be tested a lot. Each camera has its own distortion parameters as result of manufacturing process. We should test with many cameras and see the effect.